internal/reconcilers/certified_image_stream.go

```diff
@@ -44,6 +45,13 @@ func (r *CertifiedImageStreamReconciler) Reconcile(ctx context.Context, pipeline
 	stream := newImageStream(key)
 	if objects.IsObjectFound(ctx, r.Client, key, stream) {
 		log.Info("existing certified image stream found")
+
+		// setting owner reference on ImageStream CR, so CR gets garbage collected on OperatorPipeline deletion.
+		// ignoring error, since we do not need/want to requeue on this failure,
+		// and this should self correct on subsequent reconciles.
+		_ = controllerutil.SetControllerReference(pipeline, stream, r.Scheme)
```

Comment:
You need to handle this error. I would log if there's an error, and only update if there's _no_ error. In both cases, skip reconciliation to make this a no-impact operation.

Reply:
I've never seen handling this error elsewhere, are you saying we should handle this specifically because of the use case we are trying to address?

Reply:
Nah. I'm saying that handling this error gives you (the developer) more visibility into what's going wrong without having to add LoC at a later date.

`controllerutil.SetControllerReference` throws an error if applying your controller reference would conflict with an existing reference on the object. If that's happening, I would imagine you'd want to know that without having to write extra LoC just to debug it.

Reply:
I think the latest commit should address this, lmk if other changes are needed.